var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -239,7 +238,8 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         depends_on("py-pybind11@2.10.0", when="@1.13:1")
         depends_on("py-pybind11@2.6.2", when="@1.8:1.12")
         depends_on("py-pybind11@2.3.0", when="@:1.7")
-    depends_on("sleef@3.5.1_2020-12-22", when="@1.8:")
+    depends_on("sleef@3.6.0_2024-03-20", when="@2.4:")
```

Comment:
Is there a particular reason for not mapping 2.4.0 to a newer version of SLEEF?

Reply:
PyTorch is picky. All dependencies are vendored at specific commits. Historically, every time I've tried to relax these (even trying the stable release before or after), I've encountered a slew of build errors. Most dependencies don't even have stable releases.

I kinda gave up and stopped trying to convince the PyTorch devs to follow best practices. Anyone willing to 1) test ranges of versions for dependencies, 2) update PyTorch to use stable versions, 3) unvendor more dependencies, etc. is very welcome to try.

Reply:
I understand, thanks for explaining!